pkg/server/http.go

```diff
@@ -126,20 +155,32 @@ func New(opts ...HTTPOption) (*HTTP, error) {
 		http.app.Use(m)
 	}
 
+	if http.api == nil {
+		overlayAPI, err := app.New(NewNoopEngineProvider())
+		if err != nil {
+			return nil, fmt.Errorf("failed to create overlay API: %w", err)
+		}
+		http.api = overlayAPI
+	}
+
 	// Routes...
 	api := http.app.Group("/api")
 	v1 := api.Group("/v1")
 
 	// Non-Admin:
-	v1.Post("/submit", SafeHandler(overlayAPI.Commands.SubmitTransactionHandler.Handle))
-	v1.Get("/topic-managers", SafeHandler(overlayAPI.Queries.TopicManagerDocumentationHandler.Handle))
-	v1.Post("/request-foreign-gasp-node", SafeHandler(overlayAPI.Commands.RequestForeignGASPNodeHandler.Handle))
-	v1.Post("/request-sync-response", SafeHandler(overlayAPI.Commands.RequestSyncResponseHandler.Handle))
+	v1.Post("/submit", SafeHandler(http.api.Commands.SubmitTransactionHandler.Handle))
+	v1.Get("/topic-managers", SafeHandler(http.api.Queries.TopicManagerDocumentationHandler.Handle))
+	v1.Post("/request-foreign-gasp-node", SafeHandler(http.api.Commands.RequestForeignGASPNodeHandler.Handle))
+	v1.Post("/request-sync-response", SafeHandler(http.api.Commands.RequestSyncResponseHandler.Handle))
+
+	for _, r := range http.routers {
```

Comment:
Each routerHandler to register their routes. This could potentially be before the default routes which would allow a dev to bypass them. I can't decide if that's a feature or a bug.